crates/matrix-sdk-base/src/read_receipts.rs

```diff
@@ -655,4 +927,574 @@ mod tests {
         assert_eq!(receipts.num_notifications, 0);
         assert_eq!(receipts.num_mentions, 0);
     }
+
+    fn sync_timeline_message(
+        sender: &UserId,
+        event_id: impl serde::Serialize,
+        body: impl serde::Serialize,
+    ) -> SyncTimelineEvent {
+        SyncTimelineEvent::new(sync_timeline_event!({
+            "sender": sender,
+            "type": "m.room.message",
+            "event_id": event_id,
+            "origin_server_ts": 42,
+            "content": { "body": body, "msgtype": "m.text" },
+        }))
+    }
+
+    /// Smoke test for `compute_notifications`.
+    #[test]
+    fn test_basic_compute_notifications() {
+        let user_id = user_id!("@alice:example.org");
+        let other_user_id = user_id!("@bob:example.org");
+        let room_id = room_id!("!room:example.org");
+        let receipt_event_id = event_id!("$1");
+
+        let mut previous_events = Vector::new();
```

Comment:
Out of interest, do we need `eyeball_im::Vector` here? If not, I'd argue for simplicity over performance here, especially because these vectors are going to be small (right?).

Reply:
(So I'd argue for just `Vec`.)

Reply:
Oh wait, `Vector` is already in use in the production code, so this is fine. I'm still interested in why we think it's useful though.

Reply:
Yeah, we're using it to match what the sliding sync code uses, so as to avoid a temporary allocation that would be thrown away. I've tried using some generic trait with an iterator, but gave up after a bit of fighting with rustc.